gp/cartesian_graph.py

```diff
@@ -216,17 +222,21 @@ def _format_output_str_of_all_nodes(self):
             for node in active_nodes[hidden_column_idx]:
                 node.format_output_str(self)
 
-    def _fill_parameter_values(self, func_str, parameter_names_to_values):
+    def _fill_parameter_values(
+        self, func_str: str, parameter_names_to_values: Optional[Dict[str, float]] = None
+    ) -> str:
         g = re.findall("<p[0-9]+>", func_str)
-        if g and parameter_names_to_values is None:
+        if len(g) != 0 and parameter_names_to_values is None:
```

Comment:
why can't we use `if g`?

Reply:
I think it's because `if g` is ambiguous, it does not distinguish between `g = None`, `g=False`, `g=[]`, (here the latter is the case we want to catch).